cmd/cluster/logs/requests.go

```diff
@@ -0,0 +1,194 @@
+package logs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type Requester struct {
+	method      string
+	url         string
+	data        string
+	headers     map[string]string
+	successCode int
+}
+
+func (rh *Requester) send() (string, error) {
+	client := http.Client{
+		Timeout: time.Second * 10,
+	}
+
+	var req *http.Request
+	var err error
+	if rh.data != "" {
+		req, err = http.NewRequest(rh.method, rh.url, bytes.NewBuffer([]byte(rh.data)))
+	} else {
+		req, err = http.NewRequest(rh.method, rh.url, nil)
+	}
+
+	if err != nil {
+		return "", fmt.Errorf("Failed to build request %v", err)
+	}
+
+	for hdr, val := range rh.headers {
+		req.Header.Set(hdr, val)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != rh.successCode {
+		return "", fmt.Errorf("Request failed: %v", resp.Status)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+func getAccessToken() (string, error) {
+	clientSecret, err := getSecretFromVault()
+	if err != nil {
+		return "", err
+	}
+
+	reqData := url.Values{
+		"grant_type":    {"client_credentials"},
+		"scope":         {"storage:logs:read storage:buckets:read"},
+		"client_id":     {"dt0s02.5QMWDNHL"},
```

Comment:
**Question:** is this something we want publicly exposed?

Reply:
I think this got missed out and was moved directly from DT Logs. We can move this to the osdctl config.